library/src/main/java/com/schibsted/spain/barista/internal/matcher/BackgroundMatcher.kt

```diff
@@ -0,0 +1,65 @@
+package com.schibsted.spain.barista.internal.matcher
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.drawable.Drawable
+import android.support.annotation.DrawableRes
+import android.view.View
+import android.widget.ImageView
+import com.schibsted.spain.barista.internal.util.BitmapComparator
+import com.schibsted.spain.barista.internal.util.DrawableToBitmapConverter
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+
+class BackgroundMatcher private constructor(@DrawableRes private val expectedDrawableRes: Int) : TypeSafeMatcher<View>(View::class.java) {
+
+  companion object {
+    private const val EMPTY = -1
+    private const val ANY = -2
+
+    @JvmStatic
+    fun withBackground(@DrawableRes resourceId: Int) = BackgroundMatcher(resourceId)
+
+    @JvmStatic
+    fun withAnyBackground() = BackgroundMatcher(ANY)
+
+    @JvmStatic
+    fun withoutBackground() = BackgroundMatcher(EMPTY)
+  }
+
+  private var resourceName: String? = null
```

Comment:
This is only used in the `describeTo` method. It should probably be a local variable.

Reply:
But it's set at the `matchesSafely()` method. First method sets it, using the target, and the second one reads it. Check it again please.

Reply:
Mmmm can't you just do locally
```
resourceName = InstrumentationRegistry.getTargetContext().getResources().getResourceEntryName(resourceId)
```
?

Reply:
Noice! Going to do it in the new matcher and also on the old one.

Reply:
It crashes.

<img width="295" alt="screen shot 2017-11-28 at 13 00 35" src="https://user-images.githubusercontent.com/456499/33318771-2afd15dc-d43c-11e7-8f3c-61da15590d47.png">

```java
java.lang.Exception: Unexpected exception, expected<junit.framework.AssertionFailedError> but was<android.content.res.Resources$NotFoundException>
at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
at android.support.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:433)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at android.support.test.internal.runner.TestExecutor.execute(TestExecutor.java:58)
at android.support.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:375)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2022)
Caused by: android.content.res.Resources$NotFoundException: Unable to find resource ID #0xfffffffe
```

Reply:
Hmm then I guess you can't. Let's go with the current implementation then! Thanks for trying it.